### ISSUE TITLE
Adds support for <=> operator in MySQL using Python API

### ISSUE THREAD

**Issue by lujie19930803** [sqlfluff/sqlfluff#6949]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Everthing work fine when i use CLI Reference to lint my sql. But when i using Python API. It report Found unparsable section: '<=> 1'. It seems Python API still not support <=>. #6929 

### Expected Behaviour

Python API suport <=>

### Observed Behaviour

Currently Python API still not support <=>.

### How to reproduce

Run 'sqlfluff lint test.sql --config .sqlfluff ' in cmd work fine.
Use sqlfluff.lint(input_sql) return Found unparsable section: '<=> 1'

### Dialect

mysql

### Version

3.4.1

### Configuration

[sqlfluff]
dialect = mysql
templater = placeholder
max_line_length = 80
exclude_rules = references.qualification

[sqlfluff:templater:placeholder]
param_regex = @\w+\.Value

[sqlfluff:indentation]
tab_space_size = 2
indent_unit = space

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = pascal

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
The API won't automatically load any configuration files, so you'll need to specify them as a parameter with `config_path`. But there is also currently a requirement to pass the `dialect` parameter due to an issue how dialects are loaded in.

```python
import sqlfluff

input_sql = "SELECT 1 <=> 1, NULL <=> NULL, 1 <=> NULL;"
print(
    sqlfluff.lint(
        input_sql,
        dialect="mysql", # currently a bug requires this to be passed
        config_path=".sqlfluff",
    )
)
```
